toolchain/check/testdata/impl/compound.carbon

```diff
@@ -242,11 +247,13 @@ fn InstanceCallIndirect(p: i32*) {
 // CHECK:STDOUT:
 // CHECK:STDOUT: specific @G.1(constants.%Self) {
 // CHECK:STDOUT:   %Self => constants.%Self
+// CHECK:STDOUT:   %.loc13_14.1 => constants.%.3
 // CHECK:STDOUT: }
 // CHECK:STDOUT:
-// CHECK:STDOUT: specific @F.1(i32) {}
+// CHECK:STDOUT: specific @F.1(@impl.%.loc17) {}
```

Comment:
Hm, I think this should be referring to the constant value of the `FacetValue` instruction, not to the instruction itself, or we won't properly deduplicate specifics for the same type.

Probably just need a `constant_values().Get` call when we create the `FacetValue` -- or you can pass the `FacetValue` directly to `TryEvalInst` to get a `ConstantId` without creating an extra instruction.

Reply:
The code is expecting an InstId, not a ConstantId -- is the idea that I will get the InstId for the constant?